source/auth/auth.rst

```diff
@@ -106,6 +108,8 @@ Errors
 
 Drivers SHOULD raise an error as early as possible when detecting invalid values in a credential. For instance, if a ``mechanism_property`` is specified for `MONGODB-CR`_, the driver should raise an error indicating that the property does not apply.
 
+Drivers MUST raise an error if any required information for a mechanism is missing. For instance, if a ``username`` is not specified for SCRAM-SHA-256, the driver must raise an error indicating the the property is missing.
```

Comment:
There is already a paragraph below which says:

> When a user has specified a mechanism, regardless of the server version, the driver MUST honor this and attempt to authenticate.

Consider removing the `and attempt to authenticate`, because it may be impossible if all required mechanism properties are not present.

Reply:
I have removed the phrase.